libs/core/langchain_core/callbacks/manager.py

```diff
@@ -1673,7 +1672,11 @@ async def on_llm_start(
                 )
             )
 
-        await asyncio.gather(*tasks)
+        if any([h.run_inline for h in self.handlers]):
```

Comment:
If a single handler has run_inline set to True, all handlers will end up running inline, which is not the correct behavior

Reply:
good catch @eyurtsev . I've fixed it. Please let me know how the change looks.